src/zsh_completion/_firejail.in

```diff
@@ -0,0 +1,237 @@
+// Note: this looks mostly like a zsh script,
+// but this file goes through the C pre-processor once
+
+_all_firejails() {
+    local -a _all_firejails_list
+    for jail in ${(f)"$(_call_program modules_tag "firejail --list 2> /dev/null | cut -d: -f1")"}; do
+        _all_firejails_list+=${jail%% *}
+    done
+    _describe 'firejails list' _all_firejails_list
+}
+
+_all_cpus() {
+    _cpu_count=$(getconf _NPROCESSORS_ONLN)
+    for i in {0..$((_cpu_count-1))} ; do
+        print $i
+    done
+}
+
+_firejail_args=(
+    '(--profile)'{--profile=,--profile=}'[use a custom profile]: : _files -g "~/.config/firejail/*.profile"'
```

Comment:
If I type `firejail --profile=` and press tab, only directories in the current working directory are shown (no files from cwd; files from cwd will be complete after you begin to enter them) and `$HOME/.config/firejail/*.profile`.

It would be nice to show/complete files from cwd and profile names for `firejail --profile=<PROFILE-NAME> foo` (e.g. `firejail --profile=shellcheck foo`) too. Maybe we can reuse a modified version of `_profiles` and `_all_profiles` https://github.com/rusty-snake/fjp/blob/fa79d89504d425906a84590758137d1fa12b58fc/build.rs#L28-L57.

Reply:
I see what you mean. Let me look into that.

Reply:
> What's this `'(--foo)'{--foo=,foo=}` syntax, I could find anything about it?

The (--foo) means: don't show this option again once a --foo option was given.
The {--foo=,--foo=} is usually for short and long options, e.g. {-f,--foo}, but since firejail does not have those, it's twice the same. I cannot remove it and I cannot remove the 2nd argument either.
The docs at http://zsh.sourceforge.net/Guide/zshguide06.html (section 6.8.2) write correctly:

> The function _arguments has been described as having `the syntax from hell', but with the arguments already laid out in front of you it doesn't look so bad. The are three types of argument: options to _arguments itself, arguments saying how to handle options to the command (i.e. `p4 diff'), and arguments saying how to handle normal arguments to the command.